Improvement/doc zenko 1224

wabernat opened the pull request:
This PR add needed information concerning deleting objects from versioned buckets using the s3utils scripts.

Fixes Zenko 1224,

bert-e commented:
# Hello wabernatScality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get
information on this process.





*Status report is not available.*

ghost commented:
There were the following issues with this Pull Request

* Commit: 37c94e57cac34c148cf07c1b4e994aba2747dd76
  - ✖ message may not be empty
  - ✖ type may not be empty
* Commit: 12f330b96fe0e4ed780a6ca9fafb1eefe41c5490
  - ✖ message may not be empty
  - ✖ type may not be empty
* Commit: ded2619a31e5f7550baa1fd1ef5540c3ae5ee70a
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/doc-ZENKO-1224` into
`development/1.0`
* [w/1.1/improvement/doc-ZENKO-1224](https://github.com/scality/zenko/compare/development/1.1...w/1.1/improvement/doc-ZENKO-1224) will be merged into `development/1.1` (pull request #426)



The following branches will **NOT** be impacted:

* `development/0.9`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:



* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @giacomoguiulfo

* @vrancurel

* @NicolasT

* @ssalaues

ghost commented:
There were the following issues with this Pull Request

* Commit: 37c94e57cac34c148cf07c1b4e994aba2747dd76
  - ✖ message may not be empty
  - ✖ type may not be empty
* Commit: 12f330b96fe0e4ed780a6ca9fafb1eefe41c5490
  - ✖ message may not be empty
  - ✖ type may not be empty
* Commit: ded2619a31e5f7550baa1fd1ef5540c3ae5ee70a
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

bert-e commented:
# History mismatch



Merge commit #12f330b96fe0e4ed780a6ca9fafb1eefe41c5490 on the integration branch
`w/1.1/improvement/doc-ZENKO-1224` is merging a branch which is neither the current
branch `improvement/doc-ZENKO-1224` nor the development branch
`development/1.1`.

It is likely due to a rebase of the branch `improvement/doc-ZENKO-1224` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

bert-e commented:
# History mismatch



Merge commit #37c94e57cac34c148cf07c1b4e994aba2747dd76 on the integration branch
`w/1.1/improvement/doc-ZENKO-1224` is merging a branch which is neither the current
branch `improvement/doc-ZENKO-1224` nor the development branch
`development/1.1`.

It is likely due to a rebase of the branch `improvement/doc-ZENKO-1224` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

wabernat committed:
improvement: ZENKO-1224 Make S3Utils Delete script available
This change also incorporates instructions for enabling the s3utils pod.
...and a fix to one tocree declaration

ghost commented:
There were the following issues with this Pull Request

* Commit: 50a237209953e650716580baf80cc1c963b8575b
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

bert-e commented:
# History mismatch



Merge commit #af53d461d57d89adae0fdecf500468cf028a172e on the integration branch
`w/1.1/improvement/doc-ZENKO-1224` is merging a branch which is neither the current
branch `improvement/doc-ZENKO-1224` nor the development branch
`development/1.1`.

It is likely due to a rebase of the branch `improvement/doc-ZENKO-1224` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

wabernat commented:
@bert-e reset

bert-e commented:
# Lossy reset warning



There seems to be *manual* commits on integration branches (e.g. conflict
resolutions) that will be lost if you chose to *reset*.

**You can use the `force_reset` command if you still want me
to delete those branches.**

bert-e commented:
# History mismatch



Merge commit #af53d461d57d89adae0fdecf500468cf028a172e on the integration branch
`w/1.1/improvement/doc-ZENKO-1224` is merging a branch which is neither the current
branch `improvement/doc-ZENKO-1224` nor the development branch
`development/1.1`.

It is likely due to a rebase of the branch `improvement/doc-ZENKO-1224` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

wabernat commented:
This issue went stale last time around. I have simplified it and will make other changes in other tickets/builds. If there is a reason this can't be reviewed/approved, please advise.

wabernat commented:
@bert-e reset

bert-e commented:
# Lossy reset warning



There seems to be *manual* commits on integration branches (e.g. conflict
resolutions) that will be lost if you chose to *reset*.

**You can use the `force_reset` command if you still want me
to delete those branches.**

bert-e commented:
# History mismatch



Merge commit #af53d461d57d89adae0fdecf500468cf028a172e on the integration branch
`w/1.1/improvement/doc-ZENKO-1224` is merging a branch which is neither the current
branch `improvement/doc-ZENKO-1224` nor the development branch
`development/1.1`.

It is likely due to a rebase of the branch `improvement/doc-ZENKO-1224` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

wabernat commented:
@bert-e force_reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/doc-ZENKO-1224` into
`development/1.0`
* [w/1.1/improvement/doc-ZENKO-1224](https://github.com/scality/zenko/compare/development/1.1...w/1.1/improvement/doc-ZENKO-1224) will be merged into `development/1.1` (pull request #453)



The following branches will **NOT** be impacted:

* `development/0.9`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @ssalaues

* @giacomoguiulfo

* @vrancurel

* @nicolast

wabernat commented:
@bert-e approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @ssalaues

* @giacomoguiulfo

* @vrancurel

* @nicolast








*The following options are set:* **approve**

ssalaues commented on docs/Operation-Architecture/Orbit_UI/File_Operations/Deleting_Files.rst:
```diff
@@ -30,67 +33,39 @@ To delete files from a selected bucket:
+Run it as follows:
 
-    ::
+  # Exec into the pod with ``kubectl exec -it s3utils bash``
```
We don't enable this pod by default so before we exec into the pod, it has to be enable in either the options file or at run time.

This is what it would look like in the options file.
```
maintenance:
  debug:
    enabled: true
```

The other option is to have it something like below which doesn't require options file configuration and will simply instantiate and launch you into an s3utils pod.
`kubectl run s3utils --image=zenko/s3utils:0.5 -it bash`

ssalaues reviewed (changes requested):
A note about the exec line otherwise looks good to me.

ghost commented:
There were the following issues with this Pull Request

* Commit: 4be3d591ca96f7198f1148465984eeed1bd5e106
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

ghost commented:
There were the following issues with this Pull Request

* Commit: 4be3d591ca96f7198f1148465984eeed1bd5e106
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

ssalaues commented on docs/Operation-Architecture/Orbit_UI/File_Operations/Deleting_Files.rst:
```diff
@@ -30,67 +33,57 @@ To delete files from a selected bucket:
-#. To delete object versions, use the command:
+	  maintenance:
+	    debug:
+	      enabled: true
```
I forgot to add, doing it this way would also require applying an access/secret key pair.
```suggestion
	      enabled: true
	      # An access/secret key to access Zenko that will be used to configure the s3utils pod
	      accessKey: <access-key>
	      secretKey: <secret-key>
```

ssalaues commented on docs/Operation-Architecture/Orbit_UI/File_Operations/Deleting_Files.rst:
```diff
@@ -30,67 +33,57 @@ To delete files from a selected bucket:
-       ]
-     }
+	The s3utils pod is disabled by default. You can also enable it 
+	by adding the following to the options file::
```
Any change to the options file requires a `helm upgrade`
```suggestion
	by adding the following to the options file and upgrading your Zenko deployment::
```

ssalaues commented on docs/Operation-Architecture/Orbit_UI/File_Operations/Deleting_Files.rst:
```diff
@@ -30,67 +33,57 @@ To delete files from a selected bucket:
+  #. Exec into the pod with::
 
-       $ aws s3api delete-object --bucket <bucket-name> --version-id <versionID> --key <key> --endpoint-url http://<zenko.endpoint.url> --profile <aws-profile>
+     $ kubectl exec -it s3utils bash
```
Adding it via the options file will require the name to be searched for since the name can/will change (similar to looking for the cloud server manager for the InstanceID):
```suggestion
     # First grep for your s3utils pod
     $ kubectl get pods | grep s3utils
     myzenko-zenko-debug-s3utils-7f77f9b5b9-627gz   1/1  Running   0   31m
     # Then exec into the pod
     $ kubectl exec -it myzenko-zenko-debug-s3utils-7f77f9b5b9-627gz bash
```

ssalaues reviewed (commented):
With these last few clarifications, the rest makes sense and should function when followed line by line.

ghost commented:
There were the following issues with this Pull Request

* Commit: 4be3d591ca96f7198f1148465984eeed1bd5e106
  - ✖ message may not be empty
  - ✖ type may not be empty
* Commit: eda5d454678171c3c1d23c1f00123eab919c7580
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

ghost commented:
There were the following issues with this Pull Request

* Commit: 4be3d591ca96f7198f1148465984eeed1bd5e106
  - ✖ message may not be empty
  - ✖ type may not be empty
* Commit: eda5d454678171c3c1d23c1f00123eab919c7580
  - ✖ message may not be empty
  - ✖ type may not be empty
* Commit: 69ef556dc83e4fc6d665e4aec4200deddf765105
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

ghost commented:
There were the following issues with this Pull Request

* Commit: 4be3d591ca96f7198f1148465984eeed1bd5e106
  - ✖ message may not be empty
  - ✖ type may not be empty
* Commit: eda5d454678171c3c1d23c1f00123eab919c7580
  - ✖ message may not be empty
  - ✖ type may not be empty
* Commit: 69ef556dc83e4fc6d665e4aec4200deddf765105
  - ✖ message may not be empty
  - ✖ type may not be empty
* Commit: 19303150b4d99c86e10215930e2528aafac88a7a
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

bert-e commented:
# History mismatch



Merge commit #cdf9456a87626941c9215a6ec631d5a296fd8cca on the integration branch
`w/1.1/improvement/doc-ZENKO-1224` is merging a branch which is neither the current
branch `improvement/doc-ZENKO-1224` nor the development branch
`development/1.1`.

It is likely due to a rebase of the branch `improvement/doc-ZENKO-1224` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

ghost commented:
There were the following issues with this Pull Request

* Commit: c833c08b4b6ef27b1164ccb5d088036f12536b51
  - ✖ body must have leading blank line


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

ghost commented:
There were the following issues with this Pull Request

* Commit: c833c08b4b6ef27b1164ccb5d088036f12536b51
  - ✖ body must have leading blank line
* Commit: d6a1c6c29ef9a59b7e6035257cd7a679f10aec95
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

ghost commented:
There were the following issues with this Pull Request

* Commit: 9a8f423c3d30b8707028f222e9723423c3648923
  - ✖ body must have leading blank line


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

ghost commented:
There were the following issues with this Pull Request

* Commit: 9a8f423c3d30b8707028f222e9723423c3648923
  - ✖ body must have leading blank line
* Commit: a5f6da4a8a9b4b06b1bf0314976a4c614e007aca
  - ✖ message may not be empty
  - ✖ type may not be empty


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

ghost commented:
There were the following issues with this Pull Request

* Commit: 284b460ba06afcd4ca1410e21276be00e7e6641b
  - ✖ body must have leading blank line


You may need to [change the commit messages][ref] to comply with the repository contributing guidelines.

--------

🤖 This comment was generated by [**commitlint[bot]**][repo]. Please report issues [here][issues].

Happy coding!

[ref]: https://help.github.com/articles/changing-a-commit-message/
[repo]: https://github.com/z0al/commitlint-bot
[issues]: https://github.com/z0al/commitlint-bot/issues

wabernat commented:
@bert-e reset

bert-e commented:
# Lossy reset warning



There seems to be *manual* commits on integration branches (e.g. conflict
resolutions) that will be lost if you chose to *reset*.

**You can use the `force_reset` command if you still want me
to delete those branches.**





*The following options are set:* **approve**

bert-e commented:
# History mismatch



Merge commit #cdf9456a87626941c9215a6ec631d5a296fd8cca on the integration branch
`w/1.1/improvement/doc-ZENKO-1224` is merging a branch which is neither the current
branch `improvement/doc-ZENKO-1224` nor the development branch
`development/1.1`.

It is likely due to a rebase of the branch `improvement/doc-ZENKO-1224` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

wabernat commented:
@bert-e force_reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/doc-ZENKO-1224` into
`development/1.0`
* [w/1.1/improvement/doc-ZENKO-1224](https://github.com/scality/zenko/compare/development/1.1...w/1.1/improvement/doc-ZENKO-1224) will be merged into `development/1.1` (pull request #454)



The following branches will **NOT** be impacted:

* `development/0.9`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.







*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/1.0`

* :heavy_check_mark: `development/1.1`



The following branches will **NOT** be impacted:

* `development/0.9`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/1.0`

* :heavy_check_mark: `development/1.1`



The following branches have **NOT** changed:

* `development/0.9`



Please check the status of the associated issue None.

Goodbye wabernatscality.
